pkg/ibm/vpc_collection.go

```diff
@@ -352,8 +353,9 @@ func getLoadBalancers(vpcService *vpcv1.VpcV1) ([]*datamodel.LoadBalancer, error
 				return nil, err
 			}
 		}
-
-		res[i] = datamodel.NewLoadBalancer(&loadBalancers[i], listeners, pools)
+		if resourceGroupID == "" || *(loadBalancers[i].ResourceGroup.ID) == resourceGroupID {
```

Comment:
I suggest moving this check to the top of the block, and `continue` if it is does not have the right resource group. This way, you don't collect listeners and pools for skipped LBs